Mathlib/MeasureTheory/Measure/Tight.lean

```diff
@@ -0,0 +1,469 @@
+/-
+Copyright (c) 2024 Josha Dekker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne, Josha Dekker
+-/
+import Mathlib.MeasureTheory.Measure.Regular
+import Mathlib.Topology.Metrizable.Uniformity
+import Mathlib.Analysis.SpecificLimits.Basic
+
+/-!
+# (Pre-)tight measures
+The key definition of interest here is that of tight measures, `IsTight`. We first introduce two
+weaker notions, `IsSeparable` and `IsPretight`, which are equivalent on complete metric spaces. We
+provide some basic API for these notions and prove Ulam's tightness theorem
+(`of_isSeparableSpace_completeSpace`) and its strengthening `of_isSeparable_completeSpace`.
+
+## Main definitions
+* `IsSeparable`: A measure `μ` is separable if there is a separable set `S` such that
+  `μ S = μ Set.univ`.
+* `IsPretight`: A measure `μ` is pre-tight if for all `0 < ε`, there exists `K` totally bounded such
+  that `μ Kᶜ ≤ ε`.
+* `IsTight`: A measure `μ` is tight if for all `0 < ε`, there exists `K` compact such that
+  `μ Kᶜ ≤ ε`. This is defined in terms of filters. `IsTight.iff_compact_sets` establishes
```

Comment:
the lemma name is not right any more (and I'm asking you again to change it below :-).